beacon-chain/rpc/eth/light-client/handlers.go

```diff
@@ -293,72 +284,58 @@ func (s *Server) GetLightClientFinalityUpdate(w http.ResponseWriter, req *http.R
-	// Prepare
 	ctx, span := trace.StartSpan(req.Context(), "beacon.GetLightClientOptimisticUpdate")
 	defer span.End()
 
-	minSignatures := params.BeaconConfig().MinSyncCommitteeParticipants
-
-	block, err := s.getLightClientEventBlock(ctx, minSignatures)
+	block, err := s.suitableBlock(ctx, params.BeaconConfig().MinSyncCommitteeParticipants)
 	if !shared.WriteBlockFetchError(w, block, err) {
 		return
 	}
-
-	state, err := s.Stater.StateBySlot(ctx, block.Block().Slot())
+	st, err := s.Stater.StateBySlot(ctx, block.Block().Slot())
 	if err != nil {
 		httputil.HandleError(w, "could not get state: "+err.Error(), http.StatusInternalServerError)
 		return
 	}
-
-	// Get attested state
 	attestedRoot := block.Block().ParentRoot()
 	attestedBlock, err := s.Blocker.Block(ctx, attestedRoot[:])
 	if err != nil {
-		httputil.HandleError(w, "could not get attested block: "+err.Error(), http.StatusInternalServerError)
+		httputil.HandleError(w, "Could not get attested block: "+err.Error(), http.StatusInternalServerError)
 		return
 	}
 	if attestedBlock == nil {
-		httputil.HandleError(w, "attested block is nil", http.StatusInternalServerError)
+		httputil.HandleError(w, "Attested block is nil", http.StatusInternalServerError)
 		return
 	}
-
 	attestedSlot := attestedBlock.Block().Slot()
 	attestedState, err := s.Stater.StateBySlot(ctx, attestedSlot)
 	if err != nil {
-		httputil.HandleError(w, "could not get attested state: "+err.Error(), http.StatusInternalServerError)
+		httputil.HandleError(w, "Could not get attested state: "+err.Error(), http.StatusInternalServerError)
 		return
 	}
 
-	update, err := newLightClientOptimisticUpdateFromBeaconState(
-		ctx,
-		state,
-		block,
-		attestedState,
-	)
+	update, err := newLightClientOptimisticUpdateFromBeaconState(ctx, st, block, attestedState)
 	if err != nil {
-		httputil.HandleError(w, "could not get light client optimistic update: "+err.Error(), http.StatusInternalServerError)
+		httputil.HandleError(w, "Could not get light client optimistic update: "+err.Error(), http.StatusInternalServerError)
 		return
 	}
 
-	response := &structs.LightClientUpdateWithVersion{
+	response := &structs.LightClientOptimisticUpdateResponse{
 		Version: version.String(attestedState.Version()),
 		Data:    update,
 	}
 
 	httputil.WriteJson(w, response)
 }
 
-// getLightClientEventBlock - returns the block that should be used for light client events, which satisfies the minimum number of signatures from sync committee
-func (s *Server) getLightClientEventBlock(ctx context.Context, minSignaturesRequired uint64) (interfaces.ReadOnlySignedBeaconBlock, error) {
-	// Get the current state
-	state, err := s.HeadFetcher.HeadState(ctx)
+// suitableBlock returns the latest block that satisfies all criteria required for creating a new update
+func (s *Server) suitableBlock(ctx context.Context, minSignaturesRequired uint64) (interfaces.ReadOnlySignedBeaconBlock, error) {
```

Comment:
I think this name is more accurate and descriptive. Feel free to propose something better, I am not 100% happy with it

Reply:
> I think this name is more accurate and descriptive. Feel free to propose something better, I am not 100% happy with it

sounds pretty nice to me

Reply:
`suitableBlockForLightClientUpdate` sounds better tbh.

Reply:
`relevantBlock` or `suitableBlock` sound good to me.

Reply:
since this is part of the lightclient server it seems fine.